ephios/plugins/federation/models.py

```diff
@@ -183,10 +183,16 @@ def all_participations(self):
         return FederatedParticipation.objects.filter(federated_user=self.federated_user)
 
     def reverse_signup_action(self, shift):
-        return reverse("federation:shift_signup", kwargs={"pk": shift.pk})
+        return reverse(
+            "federation:shift_signup",
+            kwargs={"pk": shift.pk, "guest": self.federated_user.federated_instance.pk},
+        )
 
     def reverse_event_detail(self, event):
-        return reverse("federation:event_detail", kwargs={"pk": event.pk})
+        return reverse(
+            "federation:event_detail",
+            kwargs={"pk": event.pk, "guest": self.federated_user.federated_instance.pk},
```

Comment:
No problem here, but did you know that using
```suggestion
            kwargs={"pk": event.pk, "guest": self.federated_user.federated_instance_id},
```
causes one db query less?